### PR TITLE
LUNA-1234: iOS BpkSwapButton fix

### PR DESCRIPTION
**Pull request by LouiseReid**
Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated
- [ ] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: packages/bpk-component-swap-button/src/BpkSwapButton.module.scss [hunk 1]
```diff
@@ -19,8 +19,6 @@
 @import '../../bpk-mixins/index.scss';
 
 .bpk-swap-button {
-  width: bpk-spacing-xxl();
-  height: bpk-spacing-xxl();
```
Comment (LouiseReid) [inline packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:23]: The widths here didn't really do anything and causing issues with the width of the button. Moving these values onto the button element instead

File: packages/bpk-component-swap-button/src/BpkSwapButton.module.scss [hunk 2]
```diff
@@ -52,6 +50,10 @@
       background-color: $bpk-surface-highlight-day;
     }
 
+    span {
+      line-height: $bpk-line-height-xs;
+    }
```
Comment (LouiseReid) [inline packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:55]: Ideally would have liked to use the `withAlignment` HOC here but that was adding a margin top of 0.25rem which made shifted the svg slightly and stoped it rotating from the center

File: packages/bpk-component-swap-button/src/BpkSwapButton.module.scss [hunk 3]
```diff
@@ -61,9 +63,4 @@
       border: $bpk-border-size-xl solid $bpk-canvas-contrast-day;
     }
   }
```
Comment (LouiseReid) [inline packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:65]: Testing these changes with dev tools on a real device:

**Height and width removed from outer container**
Screenshot 2024-02-13 at 10 29 48

**Height and width set on button**
Screenshot 2024-02-13 at 10 30 02

**Span with line height of 1rem**
Screenshot 2024-02-13 at 11 21 29

Comment (LouiseReid) [reply packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:65]: Copying styles over to chrome:

Screenshot 2024-02-13 at 11 24 27

File: packages/bpk-component-swap-button/src/BpkSwapButton.module.scss [hunk 4]
```diff
@@ -35,23 +27,29 @@
 
   &__button {
     display: flex;
-    width: 100%;
-    height: 100%;
+    width: bpk-spacing-xxl();
+    height: bpk-spacing-xxl();
     justify-content: center;
     align-items: center;
+    transform: rotate(90deg);
```
Comment (LouiseReid) [inline packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:34]: previously this also had a translate Y -50%. Removed this as I think it was an overhang from this originally coming in with copied absolute styles?

All we really need here is to change the direction of the arrows for desktop vs small tablet

File: packages/bpk-component-swap-button/src/BpkSwapButton.module.scss [hunk 5]
```diff
@@ -19,9 +19,7 @@
 @import '../../bpk-mixins/index.scss';
 
 .bpk-swap-button {
-  width: bpk-spacing-xxl();
-  height: bpk-spacing-xxl();
-  transform: translateY(-50%) rotate(90deg);
+  transform: rotate(90deg);
```
[the remark this answers is not in the document]
Comment (LouiseReid) [reply packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:22]: I'd moved this around before moving it back to the container but the built storybook for the PR seems to still show it on the button. Just a caching issue?

Screenshot 2024-02-13 at 13 46 02

File: packages/bpk-component-swap-button/src/BpkSwapButton.module.scss (hunk 3 above)
Comment (LouiseReid) [reply packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:65]: **Running build locally in banana**

IMG_0857

Comment (LouiseReid) [reply packages/bpk-component-swap-button/src/BpkSwapButton.module.scss:65]: **Running build locally in banana**

Screenshot 2024-02-13 at 15 00 45

**Review by olliecurtis — APPROVED** [review]
LGTM!

**Pull request merged**